traitsui/editors/api.py

```diff
@@ -7,11 +7,15 @@
 try:
     from .array_editor import ArrayEditor
 except ImportError as e:
-    if 'numpy' not in e.message:
+    # check if failure is due to missing numpy, otherwise re-raise
+    try:
+        import numpy
+    except ImportError:
+        import warnings
+        warnings.warn('ArrayEditor is not available due to missing numpy',
+                      ImportWarning)
+    else:
```

Comment:
Yes, we do, since we can't put the `raise` in the inner `try: ... except: ...` and we want to raise if we didn't get an import error with numpy, since that implies an unexpected import error.